ttnn/cpp/ttnn/operations/conv2d.cpp

```diff
@@ -100,7 +100,7 @@ ParallelConfig determine_parallel_config(
     auto calculate_grid = [&](uint32_t num_cores_nhw) {
         if (height_sharding) {
             uint32_t cores_x_1 = num_cores_nhw >= device_grid_size[0] ? device_grid_size[0] : num_cores_nhw;
-            uint32_t cores_y_1 = (uint32_t)(num_cores_nhw / device_grid_size[0]);
```

Comment:
I don't understand what this chunk of code does. Is it just getting a `CoreRangeSet` for height sharding?

Reply:
Yeah, based on `nhw` we calculate the `num_cores_nhw` for height sharding. And above piece of code select the core grid based on `num_cores_nhw` calculated

Reply:
When nhw is less than 8 tiles. It don't properly select the core y grid.

Reply:
There's a very simple helper function that does this for you: `ttnn.num_cores_to_core_range_set`. Please switch this entire block to use this.